NPE with liberty.bootstrap property used in the pom

mvn liberty:create

```
[ERROR] Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.0.2-SNAPSHOT:create (default-cli) on project guide-getting-started: null: MojoExecutionException: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.0.2-SNAPSHOT:create (default-cli) on project guide-getting-started: null
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:137)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.NullPointerException
    at java.util.ArrayList.addAll (ArrayList.java:581)
    at io.openliberty.tools.maven.server.StartDebugMojoSupport.writeJvmOptions (StartDebugMojoSupport.java:378)
    at io.openliberty.tools.maven.server.StartDebugMojoSupport.copyConfigFiles (StartDebugMojoSupport.java:207)
    at io.openliberty.tools.maven.server.CreateServerMojo.doExecute (CreateServerMojo.java:86)
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:122)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
```

--- a/liberty-maven-plugin/src/it/server-param-pom-override-it/pom.xml
+++ b/liberty-maven-plugin/src/it/server-param-pom-override-it/pom.xml
@@ -16,6 +16,7 @@
         <!-- This maxHeap property should NOT override the one specified below in jvmOptions -->
         <liberty.jvm.minHeap>-Xms512m</liberty.jvm.minHeap>
         <liberty.jvm.maxHeap>-Xmx1024m</liberty.jvm.maxHeap>
+        <liberty.bootstrap.location>pom.xml</liberty.bootstrap.location>
         <liberty.env.JAVA_HOME>/opt/ibm/java</liberty.env.JAVA_HOME>
         <liberty.var.someVariable1>someValue1</liberty.var.someVariable1>
         <liberty.var.someVariable2>someValue2</liberty.var.someVariable2>
@@ -63,9 +64,6 @@
                         <version>${project.version}</version>
                         <type>zip</type>
                     </assemblyArtifact>
-                    <bootstrapProperties>
-                        <location>pom.xml</location>
-                    </bootstrapProperties>
                     <jvmOptions>
                         <param>-Xmx768m</param>
                     </jvmOptions>

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/StartDebugMojoSupport.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/StartDebugMojoSupport.java
@@ -314,12 +314,17 @@ public class StartDebugMojoSupport extends BasicSupport {
 
     // The properties parameter comes from the <bootstrapProperties> configuration in pom.xml and takes precedence over
     // the mavenProperties parameter, which comes from generic maven <properties> configuration.
+    // One of the passed in Maps must be not null and not empty
     private void writeBootstrapProperties(File file, Map<String, String> properties, Map<String, String> mavenProperties) throws IOException {
         if (!mavenProperties.isEmpty()) {
-            combinedBootstrapProperties = new HashMap<String,String> ();
-            // add the maven properties first so that they do not take precedence over the properties specified with <bootstrapProperties>
-            combinedBootstrapProperties.putAll(mavenProperties);
-            combinedBootstrapProperties.putAll(properties);
+            if (properties == null) {
+                combinedBootstrapProperties = mavenProperties;
+            } else {
+                combinedBootstrapProperties = new HashMap<String,String> ();
+                // add the maven properties first so that they do not take precedence over the properties specified with <bootstrapProperties>
+                combinedBootstrapProperties.putAll(mavenProperties);
+                combinedBootstrapProperties.putAll(properties);
+            }
         } else {
             combinedBootstrapProperties = properties;
         }
@@ -370,12 +375,17 @@ public class StartDebugMojoSupport extends BasicSupport {
         }
     }
 
+    // One of the passed in Lists must be not null and not empty
     private void writeJvmOptions(File file, List<String> options, List<String> mavenProperties) throws IOException {
         if (!mavenProperties.isEmpty()) {
-            combinedJvmOptions = new ArrayList<String> ();
-            // add the maven properties first so that they do not take precedence over the options specified with jvmOptions
-            combinedJvmOptions.addAll(mavenProperties);
-            combinedJvmOptions.addAll(options);
+            if (options == null) {
+                combinedJvmOptions = mavenProperties;
+            } else {
+                combinedJvmOptions = new ArrayList<String> ();
+                // add the maven properties first so that they do not take precedence over the options specified with jvmOptions
+                combinedJvmOptions.addAll(mavenProperties);
+                combinedJvmOptions.addAll(options);
+            }
         } else {
             combinedJvmOptions = options;
         }

